src/config/watcher.rs

```diff
@@ -91,26 +140,28 @@ pub fn spawn_thread<'a>(
 }
 
 fn create_watcher(
+    watcher_conf: &WatcherConfig,
     config_paths: &[PathBuf],
-) -> Result<
-    (
-        RecommendedWatcher,
-        Receiver<Result<notify::Event, notify::Error>>,
-    ),
-    Error,
-> {
+) -> Result<(Watcher, Receiver<Result<notify::Event, notify::Error>>), Error> {
     info!("Creating configuration file watcher.");
-    let (sender, receiver) = channel();
-    let mut watcher = recommended_watcher(sender)?;
-    add_paths(&mut watcher, config_paths)?;
-    Ok((watcher, receiver))
-}
 
-fn add_paths(watcher: &mut RecommendedWatcher, config_paths: &[PathBuf]) -> Result<(), Error> {
-    for path in config_paths {
-        watcher.watch(path, RecursiveMode::Recursive)?;
+    let (sender, receiver) = channel();
+    match watcher_conf {
+        WatcherConfig::RecommendedWatcher => {
+            let recommended_watcher = recommended_watcher(sender)?;
+            let mut watcher = Watcher::RecommendedWatcher(recommended_watcher);
+            watcher.add_paths(config_paths)?;
+            Ok((watcher, receiver))
+        }
+        WatcherConfig::PollWatcher(interval) => {
+            let config =
+                notify::Config::default().with_poll_interval(Duration::from_secs(*interval));
+            let poll_watcher = notify::PollWatcher::new(sender, config)?;
+            let mut watcher = Watcher::PollWatcher(poll_watcher);
+            watcher.add_paths(config_paths)?;
+            Ok((watcher, receiver))
+        }
```

Comment:
I think you can take 
```
    watcher.add_paths(config_paths)?;
            Ok((watcher, receiver))
```
out of the match branches

```suggestion
    let watcher = match watcher_conf {
        WatcherConfig::RecommendedWatcher => {
            let recommended_watcher = recommended_watcher(sender)?;
            Watcher::RecommendedWatcher(recommended_watcher)
        }
        WatcherConfig::PollWatcher(interval) => {
            let config =
                notify::Config::default().with_poll_interval(Duration::from_secs(*interval));
            let poll_watcher = notify::PollWatcher::new(sender, config)?;
            Watcher::PollWatcher(poll_watcher)
        };

     watcher.add_paths(config_paths)?;
            Ok((watcher, receiver))

```

Reply:
yeah, this seems elegant. i'll do that